packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -98,27 +80,27 @@ class MemoryController extends DisposableController
     @visibleForTesting OfflineMemoryData? offlineData,
   }) async {
     assert(!_dataInitialized.isCompleted);
-    switch (mode) {
-      case MemoryControllerCreationMode.connected:
-        await serviceConnection.serviceManager.onServiceAvailable;
-        _initializeData(
-          diffPaneController: connectedDiff,
-          profilePaneController: connectedProfile,
-        );
-      case MemoryControllerCreationMode.offlineData:
-        assert(connectedDiff == null && connectedProfile == null);
-        final loaded = await maybeLoadOfflineData(
-          ScreenMetaData.memory.id,
-          createData: createData,
-          shouldLoad: (data) => true,
-          loadData: (data) => _initializeData(offlineData: data),
-        );
-        // [maybeLoadOfflineData] will be a noop if there is no offline data for the memory screen,
-        //  so ensure we still call [_initializedData] if it has not been called.
-        assert(loaded == _dataInitialized.isCompleted);
-        if (!_dataInitialized.isCompleted) {
-          _initializeData(offlineData: offlineData);
-        }
+    if (!offlineDataController.showingOfflineData.value) {
```

Comment:
Ubernit: I'd consider swapping these cases to remove the negation for readability.